400-claims.md

```diff
@@ -46,71 +46,73 @@ This is done so that implementors can standardize on a way of relating an instal
+The CNAB claim is defined as a JSON document. The specification currently does not require that claims be formatted as Canonical JSON. Claims are immutable and are not modified after creation. When another action is executed against an installation, a new claim is created to represent the operation.
+
+The claim for the last modifying action MUST be retained. Previous claims, and/or claims for non-modifying actions, MAY be retained to provide a history of actions performed on an installation.
 
 ```json
 {
+  "action": "install",
   "bundle": {
     "credentials": {},
     "images": {},
     "invocationImages": [
       {
-        "image": "technosophos/demo2:0.2.0",
+        "image": "example/helloworld:0.1.0",
         "imageType": "docker"
       }
     ],
-    "name": "technosophos.hellohelm",
-    "outputs": {},
+    "name": "helloworld",
+    "outputs": {
+      "clientCert": {
+        "definition": "x509Certificate",
+        "path": "/cnab/app/outputs/clientCert"
+      },
+      "hostName": {
+        "applyTo": [
+          "install"
+        ],
+        "definition": "string",
+        "description": "the hostname produced installing the bundle",
+        "path": "/cnab/app/outputs/hostname"
+      },
+      "port": {
+        "definition": "port",
+        "path": "/cnab/app/outputs/port"
+      }
+    },
     "parameters": {},
     "schemaVersion": "v1.0.0",
-    "version": "0.1.0"
+    "version": "0.1.2"
   },
-  "bundleReference": "hub.example.com/my/bundle@sha256:eeeeeeeee...",
+  "bundleReference": "hub.example.com/technosophos/hellohelm@sha256:eec03d9da1bad36b3e3a4526cc774153f7024a94f25df8d2dc3ca5602fc5273d",
   "created": "2018-08-30T20:39:55.549002887-06:00",
   "custom": {},
-  "installation": "technosophos.hellohelm",
-  "modified": "2018-08-30T20:39:59.611068556-06:00",
-  "outputs": {},
+  "id": "01E5G8ZYP714JVM8NHTJQ4FH15",
+  "installation": "technosophos.helloworld",
   "parameters": {},
-  "result": {
-    "action": "install",
-    "message": "",
-    "status": "succeeded"
-  },
   "revision": "01CP6XM0KVB9V1BQDZ9NK8VP29"
 }
 ```
 Source: [400.01-claim.json](examples/400.01-claim.json)
 
 The fields above are defined as follows:
 
-- `action` (REQUIRED): The name of the action that last caused this claim to be updated. This may be any of the built-in actions (`install`, `upgrade`, `uninstall`) as well as any custom actions as defined in the bundle descriptor. The special name `unknown` MAY be used in the case where the CNAB Runtime cannot determine the action name of a claim.
+- `action` (REQUIRED): The name of the action executed against the installation. This may be any of the built-in actions (`install`, `upgrade`, `uninstall`) as well as any custom actions as defined in the bundle descriptor. The special name `unknown` MAY be used in the case where the CNAB Runtime cannot determine the action name of a claim.
```

Comment:
`unknown` has been listed in the spec as a reserved state I think since the beginning? I'm with you that I have no idea what a runtime would do with what is essentially an invalid state though. I know Porter would just end up not running anything and fail. It's not terribly useful to make it an explicit state. 😀